src/test/util.test.js

```diff
@@ -224,4 +224,60 @@ describe("Util", function () {
     });
   });
 
+  describe("deepFreeze", function () {
+
+    it("shouldn't mutate an nested object-property", function () {
+      var obj = {
+        a: {
+          b: "immutable"
+        }
+      };
+
+      Util.deepFreeze(obj);
+
+      expect(() => {
+        obj.a.b = "modified";
+      }).to.throw(TypeError);
+    });
+
+    it("shouldn't mutate an nested array-value", function () {
+      var obj = {
+        a: [
+          {},
+          "immutable"
+        ]
+      };
+
+      Util.deepFreeze(obj);
+
+      expect(() => {
+        obj.a[1] = "modified";
+      }).to.throw(TypeError);
+    });
+
+    it("shouldn't mutate first level property", function () {
+      var obj = {
+        a: 5
+      };
+
+      Util.deepFreeze(obj);
+
+      expect(() => {
+        obj.a = 8;
+      }).to.throw(TypeError);
+    });
+
+    it("can mutate first level non object", function () {
```

Comment:
We don't want that, but it's the normal behavior of Object.freeze, you can't freeze non-objects/non-arrays.